src/tzstats/tzstats_reward_provider_helper.py

```diff
@@ -70,9 +84,93 @@ def get_rewards_for_cycle(self, cycle, expected_reward = False, verbose=False):
+                delegator_info = {"staking_balance": 0, "current_balance": 0}
+                delegator_info["staking_balance"] = int(1e6 * float(delegator[idx_balance]))
+                root["delegators_balances"][delegator[idx_delegator_address]] = delegator_info
+
+        #
+        # Get current balance of delegates
+        #
+        # This is done in 2 phases. 1) make a single API call to tzscan, retrieving an array
+        # of arrays with current balance of each delegator who "currently" delegates to delegate. There may
+        # be a case where the delegator has changed delegations and would therefor not be in this array.
+        # Thus, 2) determines which delegators are not in the first result, and makes individual
+        # calls to get their balance. This approach should reduce the overall number of API calls made to tzstats.
+        #
+
+        # Phase 1
+        #
+        uri = self.api['API_URL'] + batch_current_balance_call.format(self.baking_address)
+
+        if verbose:
+            logger.debug("Requesting current balance of delegators, phase 1, {}".format(uri))
+
+        resp = requests.get(uri, timeout=5)
+
+        if verbose:
+            logger.debug("Response from tzstats is {}".format(resp.content.decode("utf8")))
+
+        if resp.status_code != 200:
+            # This means something went wrong.
+            raise ApiProviderException('GET {} {}'.format(uri, resp.status_code))
+
+        resp = resp.json()
+
+        # Will use these two lists to determine who has/has not been fetched
+        staked_bal_delegators = root["delegators_balances"].keys()
+        curr_bal_delegators = []
+        
+        for delegator in resp:
+            delegator_addr = delegator[idx_cb_delegator_address]
+            
+            # If delegator is in this batch, but has no staking balance for this reward cycle,
+            # then they must be a new delegator and are not receiving rewards at this time.
+            # We can ignore them.
+            if delegator_addr not in staked_bal_delegators:
+                continue
+
+            root["delegators_balances"][delegator_addr]["current_balance"] = int(1e6 * float(delegator[idx_cb_current_balance]))
+            curr_bal_delegators.append(delegator_addr)
+
+        # Phase 2
+        #
+        
+        # Who was not in this result?
+        need_curr_balance_fetch = list(set(staked_bal_delegators) - set(curr_bal_delegators))
+
+        # Fetch individual not in original batch
+        if len(need_curr_balance_fetch) > 0:
+
+            for d in need_curr_balance_fetch:
+            
+                uri = self.api['API_URL'] + single_current_balance_call.format(d)
+                
+                if verbose:
+                    logger.debug("Requesting current balance of delegator, phase 2, {}".format(uri))
+                
+                resp = requests.get(uri, timeout=5)
```

Comment:
A sleep command needs to be added here, otherwise the api server provider is overloaded:
```
Response from tzstats is {
  "errors": [
    {
      "code": 2000,
      "status": 502,
      "message": "bad gateway",
      "scope": "backend",
      "detail": "server is overloaded or down for maintenance"
    }
  ]
}
```

Reply:
aww. tzstats api docs say they don't throttle so i didn't include any sleeps. will add.